storage/src/mmr/mem.rs

```diff
@@ -211,6 +211,15 @@ impl<D: Digest, S: State<D>> Mmr<D, S> {
         &self.nodes[self.pos_to_index(pos)]
     }
 
+    /// Return the requested node or None if it is not stored in the MMR.
+    pub fn get_node(&self, pos: Position) -> Option<D> {
```

Comment:
If the MMR is dirty then there may be a dummy digest at position, causing the caller to silently get an invalid value